radix-engine/src/system/system_callback.rs

```diff
@@ -61,6 +64,158 @@ pub type SystemBootSubstate = SystemBoot;
+    ) -> Result<AuthModule, RejectionReason> {
+        let auth_module = match self {
+            VersionedSystemLogic::V1 => {
+                // This isn't exactly a necessary check as node logic should protect against this
+                // but keep it here for sanity
+                let intent = if executable.intents().len() != 1 {
+                    return Err(RejectionReason::TransactionNotYetSupported);
+                } else {
+                    executable.intents().get(0).unwrap()
+                };
+                AuthModule::new_with_transaction_processor_auth_zone(intent.auth_zone_init.clone())
+            }
+            VersionedSystemLogic::V2 => AuthModule::new(),
+        };
+
+        Ok(auth_module)
+    }
+
+    fn execute_transaction<Y: SystemBasedKernelApi>(
+        &self,
+        api: &mut Y,
+        executable: ExecutableTransaction,
+        global_address_reservations: Vec<GlobalAddressReservation>,
+    ) -> Result<Vec<InstructionOutput>, RuntimeError> {
+        let output = match self {
+            VersionedSystemLogic::V1 => {
+                let mut system_service = SystemService::new(api);
+                let intent = executable
+                    .intents()
+                    .get(0)
+                    .expect("This should have been checked in init");
+                let rtn = system_service.call_function(
+                    TRANSACTION_PROCESSOR_PACKAGE,
+                    TRANSACTION_PROCESSOR_BLUEPRINT,
+                    TRANSACTION_PROCESSOR_RUN_IDENT,
+                    scrypto_encode(&TransactionProcessorRunInputEfficientEncodable {
+                        manifest_encoded_instructions: intent.encoded_instructions.clone(),
+                        global_address_reservations,
+                        references: Rc::new(intent.references.clone()),
+                        blobs: intent.blobs.clone(),
+                    })
+                    .unwrap(),
+                )?;
+                let output: Vec<InstructionOutput> = scrypto_decode(&rtn).unwrap();
+                output
+            }
+            VersionedSystemLogic::V2 => {
+                let intents = executable.intents();
+                for intent in intents {
+                    let mut system_service = SystemService::new(api);
+                    let virtual_resources = intent
+                        .auth_zone_init
+                        .simulate_every_proof_under_resources
+                        .clone();
+                    let virtual_non_fungibles =
+                        intent.auth_zone_init.initial_non_fungible_id_proofs.clone();
+                    let auth_zone = AuthModule::create_auth_zone(
+                        &mut system_service,
+                        None,
+                        virtual_resources,
+                        virtual_non_fungibles,
+                    )?;
+
+                    api.kernel_set_call_frame_data(Actor::Function(FunctionActor {
```

Comment:
This doesn't seem to set it for each thread? It just keeps overriding it. Maybe I misread?